include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -480,7 +480,7 @@ class __future : private std::tuple<_Args...>
     __wait_and_get_value(sycl::buffer<_T>& __buf)
     {
         //according to a contract, returned value is one-element sycl::buffer
-        return __buf.template get_access<access_mode::read>()[0];
+        return __buf.get_host_access()[0];
```

Comment:
I am somewhat concerned that the access mode for buffers is just ignored in this patch. I think deduction tags (https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_deduction_tags) should be used to indicate the type of access. At the very least, it clearly documents the intent.

Reply:
I too agree that the deduction tags would be more clarity. But it seems like the `get_host_access` doesn't support such template deduction. https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/buffer.hpp#L572

Reply:
This function simply passes all its arguments to a constructor of `host_accessor`, and there are constructors that support deduction tags. I think the following ctor should work:
```
template <typename AllocatorT, typename TagT>
host_accessor(buffer<DataT, Dimensions, AllocatorT>& bufferRef,
              range<Dimensions> accessRange, TagT tag,
              const property_list& propList = {})
```
which should be used by `buf.get_host_access( some_tag )`. Let's try this.

Reply:
Addressed here: https://github.com/oneapi-src/oneDPL/pull/789/commits/d26ac93a6a60c810f5197c1e0c97c169a6be4f16

Reply:
Thanks for applying the update, @abagusetty. I think all feedback has been addressed.